Cabal/src/Distribution/Simple/Errors.hs

```diff
@@ -0,0 +1,61 @@
+-----------------------------------------------------------------------------
+
+-- Module      :  Distribution.Simple.Errors
+-- Copyright   :  
+-- License     :  
+--
+-- Maintainer  :  
+-- Portability :  
+--
+-- A collection of Exception Types used throughout 
+--the rest of the Cabal library package
+
+module Distribution.Simple.Errors
+(
+ CabalException (..)
+ ,exceptionCode
+ ,exceptionMessage
+) where
+
+import Distribution.Compat.Prelude
+import Distribution.Compiler
+import Distribution.Pretty
+  ( prettyShow
+  )
+
+
+-- Types representing exceptions thrown by functions in all the modules of Cabal Package
+data CabalException =  NoBenchMarkProgram FilePath
+                    | EnableBenchMark
+                    | BenchMarkNameDisable String
+                    | NoBenchMark String            
+                    | NoLibraryFound
+                    | CompilerNotInstalled CompilerFlavor
+                    | CantFindIncludeFile String
+                    | SourceDistException String
+ deriving (Show,Typeable)
+
+exceptionCode :: CabalException -> Int
+exceptionCode e = case e of 
+  (NoBenchMarkProgram _)   -> 1678

```

Comment:
A little cleaner may be to write `NoBenchMarkProgram{}` -- that way you don't need parenthesis and don't need to think about the number of constructor arguments (0, 1, 2... all become the same)